coral-trino/src/main/java/com/linkedin/coral/trino/trino2rel/StaticTrinoFunctionRegistry.java

```diff
@@ -0,0 +1,485 @@
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SameOperandTypeChecker;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlOperandTypeChecker;
+import org.apache.calcite.sql.type.SqlOperandTypeInference;
+import org.apache.calcite.sql.type.SqlReturnTypeInference;
+import org.apache.calcite.sql.type.SqlTypeFamily;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.validate.SqlUserDefinedFunction;
+
+import com.linkedin.coral.com.google.common.collect.HashMultimap;
+import com.linkedin.coral.com.google.common.collect.ImmutableList;
+import com.linkedin.coral.com.google.common.collect.Multimap;
+import com.linkedin.coral.common.functions.Function;
+import com.linkedin.coral.common.functions.FunctionRegistry;
+import com.linkedin.coral.common.functions.FunctionReturnTypes;
+import com.linkedin.coral.common.functions.GenericProjectFunction;
+import com.linkedin.coral.common.functions.OperandTypeInference;
+import com.linkedin.coral.common.functions.SameOperandTypeExceptFirstOperandChecker;
+import com.linkedin.coral.hive.hive2rel.functions.HiveExplodeOperator;
+import com.linkedin.coral.hive.hive2rel.functions.HiveFunction;
+import com.linkedin.coral.hive.hive2rel.functions.HiveJsonTupleOperator;
+import com.linkedin.coral.hive.hive2rel.functions.HiveNamedStructFunction;
+import com.linkedin.coral.hive.hive2rel.functions.HivePosExplodeOperator;
+import com.linkedin.coral.hive.hive2rel.functions.HiveRLikeOperator;
+import com.linkedin.coral.hive.hive2rel.functions.HiveReflectOperator;
+
+import static com.linkedin.coral.hive.hive2rel.functions.CoalesceStructUtility.*;
+import static org.apache.calcite.sql.fun.SqlLibraryOperators.*;
+import static org.apache.calcite.sql.fun.SqlStdOperatorTable.*;
+import static org.apache.calcite.sql.type.OperandTypes.*;
+import static org.apache.calcite.sql.type.ReturnTypes.*;
+
+
+/**
+ * Static implementation of TrinoFunctionRegistry that has hard-coded list of all
+ * function names. This has a major disadvantage that the user defined functions are
+ * not available to the registry without manually adding the entry here and uploading
+ * a new version of library.
+ * TODO: Provide function registry catalog
+ *
+ * Note that Coral maintains a copy of StaticTrinoFunctionRegistry for read only at run time.
+ * For individual query, we create a copy of function registry in a RelConTextProvider object.
+ */
+public class StaticTrinoFunctionRegistry implements FunctionRegistry {
```

Comment:
Are we planning to refactor non trino specific functions out?

Reply:
This PR has removed all the `com.linkedin.xxx` functions, and we can remove other non-trino functions and do a further code cleaning (for other classes) in the following PRs.